docs/DeveloperGuide.md

```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
-  Use case ends.
+Use case ends.
 
-* 3a. The given index is invalid.
+---
+<a name = "UC04"></a>**Use case 04: Select an insurance plan**
 
-    * 3a1. AddressBook shows an error message.
+**MSS**
 
-      Use case resumes at step 2.
+1.  User selects an insurance plan ID.
+2.  System checks if the ID is valid.
+3.  System has selected the insurance plan ID.
 
-*{More to be added}*
+Use case ends.
+
+**Extensions**
+
+* 3a. The client does not have the specified insurance plan.
+    * 3a1. System shows an error message to user.
+
+    Use case ends.
+
+* 3b. The insurance plan id is invalid.
+    * 3b1. System shows an error message to user.
+
+    Use case ends.
+---
+**Use case 05: Add an insurance plan to a client**
+
+**MSS**
+
+1.  User requests to list clients.
+2.  System shows a list of clients.
+3.  User [selects a client](#UC02) and [selects an insurance plan](#UC04) to be added to the specified client.
+4.  System adds the insurance plan to the client.
+
+Use case ends.
+
+**Extensions**
+
+* 3a. The client already has the given insurance plan.
+
+    * 3a1. System shows an error message.
+
+    Use case ends.
+---
+**Use case 06: Remove an insurance plan to a client**
+
+**MSS**
+
+1.  User requests to list clients.
+2.  System shows a list of clients.
+3.  User [selects a client](#UC02), [selects an insurance plan](#UC04) of that client and requests it to be deleted.
+4.  System removes the insurance plan from the client.
+
+Use case ends.
+
+**Extensions**
+
+* 3a. The client does not have the specified insurance plan.
+    * 3a1. System shows an error message to user.
+
+    Use case ends.
+---
+**Use case 07: View all claims of a client**
+
+**MSS**
+
+1.  User [selects a client](#UC02) and requests to view all claims of that client.
+2.  System shows all claims for the client.
+
+Use case ends.
+
+**Extensions**
+
+* 2a. There are no claims for the client.
+    * 2a1. An appropriate message about the client having no claims will be shown.
+
+    Use case ends.
+---
+<a name = "UC08"></a>**Use case 08: Select a claim of a client**
+
+**MSS**
+
+1.  User [selects a client](#UC02), [selects a insurance plan](#UC04) of the client and requests to select a claim of that specific insurance plan.
+2.  System selects the claim specified by the user.
+
+Use case ends.
+
+**Extensions**
+
+* 1a. Claim ID is invalid.
+    * 1a1. An appropriate error message will be shown.
+
+    Use case ends.
+---
+**Use case 09: Add a claim to a client**
+
+**MSS**
+
+1.  User [selects a client](#UC02) and [an insurance plan](#UC04) of that client and requests to add a claim to selected insurance plan.
+2.  System adds the claim to the insurance plan.
+
+Use case ends.
+
+**Extensions**
+
+* 3a. The claim already exists for the client.
+
+    * 3a1. System shows an error message.
+
+    Use case ends.
+---
+**Use case 09: Close a claim for a client**
+
+**MSS**
+
+1.  User [selects a claim](#UC08) requests to close that claim for the specified client.
```

Comment:
Missing "select a client" here, can edit it as:
1. User [selects a client](#UC02), [selects a claim](#UC08) and requests to close that claim for the specified client.